cruce.cpp

```diff
@@ -0,0 +1,100 @@
+#include<fstream>
+using namespace std;
+ifstream fin("cruce.in");
+ofstream fout("cruce.out");
+long long columnSum[501]={0};
+long long lineSum[501]={0};
+void read(int &n, int &k, int matrix[501][501])
+{
+    fin>>n>>k;
+    for(int index=1; index<=n; ++index)
+    {
+        for(int next=1; next<=n; ++next)
+        {
+            fin>>matrix[index][next];
+            lineSum[index]+=matrix[index][next];
+            columnSum[next]+=matrix[index][next];
+        }
+    }
+}
+
+
+
+void solveTask2(int k, int n, int matrix[501][501])
+{
+    int colIndex = 0, lineIndex = 0;
+    long long currentSum=0;
+    long long globalSum=0;
+    //solving for the lines first
+    int current_start=1, current_end=k;
+    
+
+    currentSum=lineSum[1]+lineSum[k];
+    while(current_end<=n)
+    {
+        if(currentSum>globalSum)
```

Comment:
Trebuie sa consideri cazul inc are `currentSum==globalSum`. Trebuie sa tii minte *toti* indicii care dau suma maxima, si apoi sa vezi care din ei este cel mai bun cand aduni coloanele.

Atentie cand aduni coloanele sa nu aduni de doua ori numerele care sunt in "mijlocul" crucii (adica si pe linie si pe coloana).